proxy/certs/certs.go

```diff
@@ -34,9 +34,43 @@ import (
 	sqladmin "google.golang.org/api/sqladmin/v1beta4"
 )
 
+const defaultUserAgent = "custom cloud_sql_proxy version >= 1.10"
+
 // NewCertSource returns a CertSource which can be used to authenticate using
-// the provided oauth token. The provided client must not be nil.
+// the provided client, which must not be nil.
+//
+// This function is deprecated; use NewCertSourceOpts instead.
 func NewCertSource(host string, c *http.Client, checkRegion bool) *RemoteCertSource {
+	return NewCertSourceOpts(c, RemoteOpts{
+		Host:         host,
+		IgnoreRegion: !checkRegion,
+		UserAgent:    defaultUserAgent,
+	})
+}
+
+// RemoteOpts are a collection of options for NewCertSourceOpts. All fields are
+// optional.
+type RemoteOpts struct {
+	// Host specifies the base path for the sqladmin API. If left blank, the
+	// default from the autogenerated sqladmin library is used.
+	Host string
```

Comment:
Out of scope of this CL: This ought to be called something other than 'host'. Maybe APIBasePath?